pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,12 +70,138 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 			}
 		}
 	}
+	return nil
+}
 
-	instance.Status.StorageClassesCreated = true
-
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalCluster(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResource
+	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	// storageClasses stores all the StorageClasses coming from the JSON
+	var storageClasses map[string]map[string]string = make(map[string]map[string]string)
+
+	for _, d := range data {
+		objectMeta := metav1.ObjectMeta{
+			Name:            d.Name,
+			Namespace:       instance.Namespace,
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		}
+		objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: objectMeta,
+				Data:       d.Data,
+			}
+			found := &corev1.ConfigMap{}
```

Comment:
Wouldn't that be always true? The `found` cm needs to contain the `objectMeta`.

Reply:
True, Addressed